src/librustc/middle/dead.rs

```diff
@@ -464,21 +460,19 @@ impl<'a, 'tcx> DeadVisitor<'a, 'tcx> {
         // method of a private type is used, but the type itself is never
         // called directly.
         let impl_items = self.tcx.impl_items.borrow();
-        match self.tcx.inherent_impls.borrow().get(&self.tcx.map.local_def_id(id)) {
-            None => (),
-            Some(impl_list) => {
+        if let Some(impl_list) =
+            self.tcx.inherent_impls.borrow().get(&self.tcx.map.local_def_id(id)) {
                 for impl_did in impl_list.iter() {
                     for item_did in impl_items.get(impl_did).unwrap().iter() {
                         if let Some(item_node_id) =
-                                self.tcx.map.as_local_node_id(item_did.def_id()) {
-                            if self.live_symbols.contains(&item_node_id) {
-                                return true;
+                            self.tcx.map.as_local_node_id(item_did.def_id()) {
+                                if self.live_symbols.contains(&item_node_id) {
+                                    return true;
+                                }
                             }
```

Comment:
The previous comment (on src/libsyntax/print/pprust.rs, line 3020) makes me think you might mean
```rust
                        if let Some(item_node_id) =
                            self.tcx.map.as_local_node_id(item_did.def_id()) {
                             if self.live_symbols.contains(&item_node_id) {
                                 return true;
                             }
                        }
```
(with just one extra space on the innermost level, as if the rationale is that this is just enough to differentiate it from the enclosing multi-line if-let condition, without wasting an entire extra four-space level of indentation), unless it's this comment that demonstrates the correct rule?